docs/zkasm/infra.md

```diff
@@ -0,0 +1,64 @@
+# General Overview
+
+Firstly, we aim to implement the following types of tests and benchmarks:
+- Wasm spectest: Programs generated from Wasm spectest.
+- Benchmarks: Programs we manually write in Wasm or Rust and compile to Wasm.
+
+Both types of programs will undergo the same pipeline:
+- Compile to zkasm.
+- Run the generated zkasm & measure performance.
+- Track changes.
+- Alert in GitHub PRs in case of degradation.
+
+We use the snapshot-testing pattern in many places (link).
+
+Performance data is collected from both tests and benchmarks.
+Benchmarks also serve as global tests.
+
+## Test Creation and Generation
+We use a snapshot-based pattern for generating Wasm from spectest wast files. Benchmarks are added manually.
+
+Our goal is to simplify adding benchmarks (simply add a `.wat` file to the benchmarks directory, and it automatically goes through the pipeline) and to automatically pull all updates from Wasm spectests.
+
+## Wasm Compiling
+We aim to have a single test that compiles all `.wat` files to zkasm and writes results to a CSV file in a snapshot-based pattern. This script should be capable of:
```

Comment:
What do you mean by "single test" here? A single Rust test case? Or a single shell script to run tests?